test/python/circuit/test_random_circuit.py

```diff
@@ -160,3 +163,196 @@ def test_random_circuit_with_zero_distribution(self):
+        # Do not change the function signature.
+        conditions = (getattr(instruction.operation, "condition", None) for instruction in circ)
+        conditions = [x for x in conditions if x is not None]
+        self.assertNotEqual(conditions, [])
+        for register, value in conditions:
+            self.assertIsInstance(register, (ClassicalRegister, Clbit))
+            # Condition values always have to be Python bigints (of which `bool` is a subclass), not
+            # any of Numpy's fixed-width types, for example.
+            self.assertIsInstance(value, int)
+
+    def test_random_mid_circuit_measure_conditional(self):
+        """Test random circuit with mid-circuit measurements for conditionals."""
+        qc = random_circuit_from_graph(
+            interaction_graph=self.interaction_graph,
+            min_2q_gate_per_edge=1,
+            measure=True,
+            conditional=True,
+            reset=True,
+            seed=0,
+        )
+        dag = circuit_to_dag(qc)
+
+        # Before a condition, there needs to be measurement of atleast one of the qubits.
+        measure_at = None
+        condition_at = None
+        for qubit_idx, wire in enumerate(dag.wires):
+            if condition_at is not None and measure_at is not None:
+                break
+            for layer_num, dag_op_node in enumerate(dag.nodes_on_wire(wire, only_ops=True)):
+                if condition_at is None and getattr(dag_op_node.op, "condition", None) is not None:
+                    condition_at = {"qubit_idx": qubit_idx + 1, "layer_no": layer_num + 1}
+                elif measure_at is None and dag_op_node.op.name == "measure":
+                    measure_at = {"qubit_idx": qubit_idx + 1, "layer_no": layer_num + 1}
+
+        self.assertGreater(condition_at["layer_no"], measure_at["layer_no"])
+
+    def test_2q_gates_applied_to_edges_from_interaction_graph(self):
+        """Test 2Q gates are applied to the qubit-pairs given by the interaction graph supplied"""
+        qc = random_circuit_from_graph(
+            interaction_graph=self.interaction_graph,
+            min_2q_gate_per_edge=1,
+            measure=True,
+            conditional=True,
+            reset=True,
+            seed=0,
+        )
+        dag = circuit_to_dag(qc)
+
+        cp_mp = set()
+        pydi_graph, _, _, _ = self.interaction_graph
+        edge_list = set(pydi_graph.edge_list())
+
+        for wire in dag.wires:
+            for dag_op_node in dag.nodes_on_wire(wire, only_ops=True):
+                if dag_op_node.op.num_qubits == 2:
+                    control, target = dag_op_node.qargs
+                    control_idx = control._index
+                    target_idx = target._index
+                    cp_mp.update({(control_idx, target_idx)})
+
+        # make sure every qubit-pair from the circuit actually present in the edge_list
+        for cp in cp_mp:
+            self.assertTrue(cp in edge_list)
```

Comment:
You can replace this chunk of code by `build_interaction_graph` in https://github.com/Qiskit/qiskit/blob/main/qiskit/transpiler/passes/layout/vf2_utils.py and a subgraph isomorphism check. :-)

Reply:
I have kept the logic untouched, because for some reasons `build_interaction_graph` is giving out a coupling map, which is not correct for the given circuit.
On visual inspection, it was confirmed that the coupling map given out by this logic is indeed correct and serve the purpose.